src/python/test/test_dxpy.py

```diff
@@ -661,6 +661,68 @@ def test_part_splitting(self):
         self.assertEquals(parts['1']['size'], 5242880)
         self.assertEquals(parts['2']['size'], 2952504)
 
+class TestFolder(unittest.TestCase):
+
+    def setUp(self):
+        setUpTempProjects(self)
+        self.temp_dir = tempfile.mkdtemp(prefix="dx-toolkit.dxpy.TestFolder.")
+        self.temp_file_fd, self.temp_file_path = tempfile.mkstemp(prefix="dx-toolkit.dxpy.TestFile.")
+        with os.fdopen(self.temp_file_fd, 'w') as temp_file:
+            temp_file.write('42')
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+        os.remove(self.temp_file_path)
+        tearDownTempProjects(self)
+
+    def test_download_folder(self):
+        dxproject = dxpy.DXProject(self.proj_id)
+
+        # Creating remote folders
+        dxproject.new_folder("/a/b/c/d", parents=True)
+        dxproject.new_folder("/a/e/f/g", parents=True)
+        dxproject.new_folder("/h/i/j/k", parents=True)
+
+        # Filling remote folders with objects
+        path = []
+        i = 1
+        for f in ["/", "a", "b", "c", "d"]:
```

Comment:
When you want to iterate over a collection and also count the number of items you've seen in parallel, you can use `enumerate`:

```
for i, f in enumerate(["/", "a", "b", "c", "d"], start=1):
```